src/server_manager/web_app/ui_components/outline-region-picker-step.ts

```diff
@@ -146,14 +158,27 @@ export class OutlineRegionPicker extends LitElement {
     this.selectedLocationId = inputEl.value;
   }
 
+  _flagImage(item: DisplayLocation): string {
+     return FLAG_MAPPING[item.name?.getCountryCode()] || `${FLAG_IMAGE_DIR}/unknown.png`;
+  }
+
+  _selectedLocation(): DisplayLocation {
+    return this.locations.find(item => item.id === this.selectedLocationId);
+  }
+
   _handleCreateServerTap(): void {
     this.isServerBeingCreated = true;
     const params = {
       bubbles: true,
       composed: true,
-      detail: {selectedRegionId: this.selectedLocationId}
+      detail: {selectedLocation: this._selectedLocation()}
```

Comment:
I'm deliberately passing the object (now a `CloudLocation`) rather than the ID.  Otherwise, we will need the UI code to reconstruct the `CloudLocation`, using a cloud-specific DataCenterId-to-GeoId conversion function.